### PR TITLE
Disable/Enable downloading videos/audio over mobile networks

### PR DESCRIPTION
**Pull request by CHRINIMUE**
I have added a new Setting "Mobile Downloads". If you allow this you will be able to download videos or audio with the mobile network, roaming and metered (counted) network, if you disable (default) is it will download it only over the wifi. A lite feature for #247. Not complete, but a part ...

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java [hunk 1]
```diff
@@ -182,6 +184,33 @@ private void download(String url, String title,
...
+            } else {
+                request.setAllowedNetworkTypes(DownloadManager.Request.NETWORK_WIFI);//enabel wifi
+                request.setAllowedOverRoaming(false);                                //disable roaming
+                if (Build.VERSION.SDK_INT >= 16){
+                    request.setAllowedOverMetered(false);                            //Set counted/metered Netowrk as allowed
+                }
+            }
+
+            /*
+            if (mna) {
```
Comment (CHRINIMUE) [inline app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java]: UPS, i forgot to remove this comment :-/
I will fix it ;-)

**Comment by theScrabi** [comment]
Nice.

**Comment by theScrabi** [comment]
Hä why did i close this? Sorry.

**Comment by CHRINIMUE** [comment]
OK, now I have deleted the unnecessary code. Is it OK, or should I do some changes?

**Comment by CHRINIMUE** [comment]
Now I know how to add a limit for mobile network usage. I will add it in a few days. But I will not be able to test these function, are there volunteers? And should I open a new "Settings Page" or "Settings Category" for "Data usage" or something like that?

**Comment by satiricon** [comment]
Oops. I'm heavily modifying this file to allow for a few choices before starting the downloading process (or processes). Hopefully will not be too hard to merge both features.

**Comment by CHRINIMUE** [comment]
OK, it will make no sense to merge these commit, because my part will be deleted by your merge (@satiricon, #285). And your merge is very good and important to add functions to the download.
[HERE](https://github.com/theScrabi/NewPipe/pull/285/files#diff-ab1439b1792279a116a666f3f2274e7eL179), these code will be deleted including mine. So I will close this, because I don't know how tho do the same with the HTTPurlConnection.

**Comment by satiricon** [comment]
I'm pretty sure that I can somehow include your code into my pull request. Give me a while and I'll do it.

**Pull request closed without merging**